crates/starknet/types/src/lib.rs

```diff
@@ -85,3 +85,51 @@ impl From<Call> for starknet::core::types::Call {
         }
     }
 }
+
+/// Validates that a Felt value represents a valid Starknet contract address.
+///
+/// ### Arguments
+/// * `felt` - The Felt value to validate as a Starknet address
+///
+/// ### Returns
+/// Returns `true` if the `felt` is a valid Starknet address, `false` otherwise.
+pub fn is_valid_starknet_address(felt: &Felt) -> bool {
+    felt > &Felt::from(2u64)
+        && felt
+            < &Felt::from_hex_unchecked(
+                "0x800000000000000000000000000000000000000000000000000000000000000",
+            )
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
this is a good unites, but it doesn't guarantee that the check is actually done in the rest of the codebase.
To do so, add an integration test in `crates/tests/node-tests/`.
For this just create a new test file and add it to `crates/tests/node-tests/Cargo.toml` same way as other. In it, just call the melt endpoint with some bad payee values, and check that you indeed get an error.

Same with `crates/tests/signer-tests/`